lib/src/tasks/gen_test_runner/api.dart

```diff
@@ -0,0 +1,131 @@
+// Copyright 2015 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+library dart_dev.src.tasks.gen_test_runner.api;
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:dart_dev/src/tasks/gen_test_runner/config.dart';
+import 'package:dart_dev/src/tasks/task.dart';
+
+Future<GenTestRunnerTask> genTestRunner(TestRunnerConfig currentConfig) async {
+  var taskTitle = 'gen-test-runner';
+  var args = ['-d ${currentConfig.directory}', '-e ${currentConfig.env}'];
+  currentConfig.genHtml ? args.add('--genHtml') : args.add('--no-genHtml');
+
+  GenTestRunnerTask task =
+      new GenTestRunnerTask('$taskTitle ${args.join(' ')}');
+
+  File generatedRunner =
+      new File('${currentConfig.directory}/${currentConfig.filename}.dart');
+  IOSink writer = generatedRunner.openWrite(mode: FileMode.WRITE);
+
+  Directory testDirectory = new Directory(currentConfig.directory);
+  List<File> testFiles = [];
+  List<FileSystemEntity> allFiles =
+      testDirectory.listSync(recursive: true, followLinks: false);
+  allFiles.forEach((FileSystemEntity entity) {
+    if (entity is File) {
+      if (entity.path.endsWith('_test.dart') &&
+          !entity.path.endsWith('${currentConfig.filename}.dart')) {
```

Comment:
nit - since this condition is used twice, might as well do it once and hold the result in a bool for use in the 2 if statements